### PR TITLE
Extract actor info from public commands and pass it to handlers

### PR DESCRIPTION
**Pull request by Rotorsoft**
First draft

### CODE REVIEW

File: libs/eventually/src/app.ts [hunk 1]
```diff
@@ -94,38 +94,38 @@ export abstract class AppBase extends Builder implements Reader {
...
+      : { event: undefined, count: 0 };
+    const events = await callback(snapshot.state);
     events.map((event) => this._validate(event));
     if (streamable) {
       const committed = await store().commit(
         streamable.stream(),
         events,
         metadata,
-        expectedVersion,
+        metadata.causation.command?.expectedVersion,
```
Comment (tomasruizr) [inline libs/eventually/src/app.ts:124]: what is the benefit of storing the expected version in the command causation?, if it got to the store, we can safely assume the expected version was right or not relevant... am I missing something?

**Review by tomasruizr — COMMENTED** [review]
This looks great, elegant and efficient!

File: libs/eventually-express/src/ExpressApp.ts [hunk 2]
```diff
@@ -250,7 +253,7 @@ export class ExpressApp extends AppBase {
     // ensure catch-all is last handler
     this._app.use(
       // eslint-disable-next-line
-      (error: Error, req: Request, res: Response, next: NextFunction) => {
+      (error: Error, _: Request, res: Response, __: NextFunction) => {
```
[the remark this answers is not in the document]
Comment (Rotorsoft) [reply libs/eventually-express/src/ExpressApp.ts:256]: _ to represent positional arguments that are not used, just a convention... but important for this function signature to work

File: libs/eventually/src/types.ts [hunk 3]
```diff
@@ -46,29 +46,37 @@ export type Message<Name extends string, Type extends Payload> = {
...
 };
 
+/**
+ * Actors invoke public commands
+ * - `name` Actor name
+ * - `roles` Actor roles
+ */
+export type Actor = {
+  name: string;
+  roles: string[];
```
[the remark this answers is not in the document]
Comment (Rotorsoft) [reply libs/eventually/src/types.ts:56]: Not sure granular permissions should be part of the framework... I tend to think role based is good enough and you can always derive granularity in your application from roles

File: services/accounts/src/__tests__/happy.spec.ts [hunk 4]
```diff
@@ -67,7 +67,7 @@ describe("happy path", () => {
 
     // when
     await app().event(policies.IntegrateAccount2, t);
-    await sleep(100);
+    await sleep(1000);
```
[the remark this answers is not in the document]
Comment (Rotorsoft) [reply services/accounts/src/__tests__/happy.spec.ts:70]: just a sign of paranoia 😆 ... these tests had random failures and this seems to fix it

File: libs/eventually/src/app.ts (hunk 1 above)
Comment (Rotorsoft) [reply libs/eventually/src/app.ts:124]: this is just to audit when the command was invoked with expectedVersion or not

**Pull request merged**